Expo-camera and expo-contacts throw package org.jetbrains.annotations does not exist

### Summary

Note: This issue is consistenlty reproducible on EAS builds and local Expo builds

Using expo-camera with expo-contacts consistently fails android builds with `package org.jetbrains.annotations does not exist`. Please see https://github.com/sshquack/boring-app-01 for full repro code.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

41

### Environment

```
  Expo CLI 4.5.0 environment info:
    System:
      OS: macOS 10.15.7
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 14.16.1 - ~/.n/bin/node
      Yarn: 1.22.10 - ~/.n/bin/yarn
      npm: 7.15.0 - ~/.n/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    IDEs:
      Android Studio: 4.2 AI-202.7660.26.42.7351085
      Xcode: 12.4/12D4e - /usr/bin/xcodebuild
    npmPackages:
      expo: ~41.0.1 => 41.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: ~0.63.4 => 0.63.4 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.5.0
    Expo Workflow: bare
```

### Reproducible demo or steps to reproduce from a blank project

1. Create new expo project
```shell
expo init --npm -t expo-template-blank-typescript
```

2. Eject to bare workflow
```shell
expo eject --npm
```

3. Install `expo-camera` and `expo-contacts`
```shell
expo install expo-camera expo-contacts
```

4. Add permissions to `AndroidManifest.xml`
```shell
  <uses-permission android:name="android.permission.CAMERA"/>
  <uses-permission android:name="android.permission.READ_CONTACTS" />
  <uses-permission android:name="android.permission.WRITE_CONTACTS" />
```

5. Follow extra steps for bare workflow https://github.com/expo/expo/tree/master/packages/expo-camera
```shell
allprojects {
    repositories {
        mavenLocal()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url("$rootDir/../node_modules/react-native/android")
        }
        maven {
            // Android JSC is installed from npm
            url("$rootDir/../node_modules/jsc-android/dist")
        }

        google()
        jcenter()
        maven { url 'https://www.jitpack.io' }

        maven {
          // expo-camera bundles a custom com.google.android:cameraview
          url "$rootDir/../node_modules/expo-camera/android/maven"
        }
    }
}
```

6. Build for android
```shell
react-native run-android
```

7. Build fails with error `/code/expo/boring-app/node_modules/expo-contacts/android/src/main/java/expo/modules/contacts/Contact.java:14: error: package org.jetbrains.annotations does not exist`
```bash
> react-native run-android

info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
Jetifier found 1310 file(s) to forward-jetify. Using 8 workers...
info JS server already running.
info Installing the app...
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :app

Installing unimodules:
 unimodules-core@7.1.0 from /code/expo/boring-app/node_modules/@unimodules/core
 unimodules-react-native-adapter@6.2.2 from /code/expo/boring-app/node_modules/@unimodules/react-native-adapter
 expo-application@3.1.2 from /code/expo/boring-app/node_modules/expo-application
 unimodules-permissions-interface@6.2.0 from /code/expo/boring-app/node_modules/expo-camera/node_modules/unimodules-permissions-interface
 expo-camera@11.0.3 from /code/expo/boring-app/node_modules/expo-camera
 expo-constants@10.1.3 from /code/expo/boring-app/node_modules/expo-constants
 expo-contacts@9.1.2 from /code/expo/boring-app/node_modules/expo-contacts
 expo-error-recovery@2.1.0 from /code/expo/boring-app/node_modules/expo-error-recovery
 expo-file-system@11.0.2 from /code/expo/boring-app/node_modules/expo-file-system
 expo-font@9.1.0 from /code/expo/boring-app/node_modules/expo-font
 expo-image-loader@2.1.1 from /code/expo/boring-app/node_modules/expo-image-loader
 expo-keep-awake@9.1.2 from /code/expo/boring-app/node_modules/expo-keep-awake
 expo-permissions@12.0.1 from /code/expo/boring-app/node_modules/expo-permissions
 expo-splash-screen@0.10.2 from /code/expo/boring-app/node_modules/expo-splash-screen
 expo-structured-headers@1.0.1 from /code/expo/boring-app/node_modules/expo-structured-headers
 expo-updates@0.5.5 from /code/expo/boring-app/node_modules/expo-updates
 unimodules-app-loader@2.1.0 from /code/expo/boring-app/node_modules/unimodules-app-loader
 unimodules-barcode-scanner-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-barcode-scanner-interface
 unimodules-camera-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-camera-interface
 unimodules-constants-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-constants-interface
 unimodules-face-detector-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-face-detector-interface
 unimodules-file-system-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-file-system-interface
 unimodules-font-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-font-interface
 unimodules-image-loader-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-image-loader-interface
 unimodules-sensors-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-sensors-interface
 unimodules-task-manager-interface@6.1.0 from /code/expo/boring-app/node_modules/unimodules-task-manager-interface

Found some duplicated unimodule packages. Installed the ones with the highest version number.
Make sure following dependencies of your project are resolving to one specific version:
 unimodules-permissions-interfacem 1s]

> Task :expo-file-system:processDebugManifest
/code/expo/boring-app/node_modules/expo-file-system/android/src/main/AndroidManifest.xml:9:9-20:20 Warning:
        provider#expo.modules.filesystem.FileSystemFileProvider@android:authorities was tagged at AndroidManifest.xml:9 to replace other declarations but no other declaration present

> Task :app:processDebugMainManifest
/code/expo/boring-app/android/app/src/debug/AndroidManifest.xml:17:9-26:20 Warning:
        provider#expo.modules.filesystem.FileSystemFileProvider@android:authorities was tagged at AndroidManifest.xml:17 to replace other declarations but no other declaration present

> Task :unimodules-core:compileDebugJavaWithJavac

> Task :expo-application:compileDebugJavaWithJavac

> Task :expo-file-system:compileDebugJavaWithJavac

> Task :expo-camera:compileDebugJavaWithJavac

> Task :expo-constants:compileDebugJavaWithJavac

> Task :expo-contacts:compileDebugJavaWithJavac FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings
355 actionable tasks: 355 executed
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: /code/expo/boring-app/node_modules/expo-application/android/src/main/java/expo/modules/application/ApplicationModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /code/expo/boring-app/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /code/expo/boring-app/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /code/expo/boring-app/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsService.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/code/expo/boring-app/node_modules/expo-contacts/android/src/main/java/expo/modules/contacts/Contact.java:14: error: package org.jetbrains.annotations does not exist
import org.jetbrains.annotations.Nullable;
                                ^
/code/expo/boring-app/node_modules/expo-contacts/android/src/main/java/expo/modules/contacts/Contact.java:192: error: cannot find symbol
  @Nullable
   ^
  symbol:   class Nullable
  location: class Contact
Note: /code/expo/boring-app/node_modules/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':expo-contacts:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4m 3s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup. Run CLI with --verbose flag for more details.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
Note: Some input files use unchecked or unsafe operations.

Process finished with exit code 1
```

--- a/packages/expo-contacts/CHANGELOG.md
+++ b/packages/expo-contacts/CHANGELOG.md
@@ -9,7 +9,7 @@
 ### 🐛 Bug fixes
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
-- Switch to androidx.annotation.Nullable.
+- Switch to androidx.annotation.Nullable. ([#13133](https://github.com/expo/expo/pull/13133) by [@brentvatne](https://github.com/brentvatne))
 
 ### 💡 Others
 

--- a/packages/expo-contacts/CHANGELOG.md
+++ b/packages/expo-contacts/CHANGELOG.md
@@ -9,6 +9,7 @@
 ### 🐛 Bug fixes
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
+- Switch to androidx.annotation.Nullable.
 
 ### 💡 Others
 

--- a/packages/expo-contacts/android/build.gradle
+++ b/packages/expo-contacts/android/build.gradle
@@ -76,5 +76,6 @@ dependencies {
   unimodule "unimodules-core"
   unimodule 'expo-modules-core'
 
+  implementation 'androidx.annotation:annotation:1.2.0'
   implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${safeExtGet('kotlinVersion', '1.4.21')}"
 }

--- a/packages/expo-contacts/android/src/main/java/expo/modules/contacts/Contact.java
+++ b/packages/expo-contacts/android/src/main/java/expo/modules/contacts/Contact.java
@@ -11,7 +11,7 @@ import android.provider.ContactsContract;
 import android.text.TextUtils;
 import android.util.Log;
 
-import org.jetbrains.annotations.Nullable;
+import androidx.annotation.Nullable;
 
 import java.io.ByteArrayOutputStream;
 import java.text.ParseException;

